rosidl_typesupport_opensplice_cpp/include/rosidl_typesupport_opensplice_cpp/requester.hpp

```diff
@@ -137,10 +139,8 @@ class Requester
       estr = "DomainParticipant::create_topic: failed for response";
       goto fail;
     }
-
-    // Let OpenSplice do any needed encoding
     content_filtered_response_topic_ = participant_->create_contentfilteredtopic(
-      service_name_.c_str(), response_topic_,
+      content_filtered_topic_name.c_str(), response_topic_,
```

Comment:
When I wrote a test case with multiple clients listening on the same topic, the case failed for Opensplice and passed both Connext implementations. After some debugging it was apparent that I couldn't have multiple ContentFilteredTopic objects with the same name (which was happening because I had multiple clients passing the same service name to the ContentFilteredTopic objects). So I changed the topic name to `ServiceName_writer_guid.first_writer_guid.second` (as you can see above) so that each client would have a unique ContentFilteredTopic name.